test/t/unit/test_unit_known_hosts.py

```diff
@@ -127,6 +126,8 @@ def test_included_configs(self, bash, hosts):
         expected.extend("asterisk_1 asterisk_2".split())
         # fixtures/_known_hosts/.ssh/config_question_mark
         expected.append("question_mark")
+        # fixtures/_known_hosts/.ssh/config_comment
+        expected.append("with_comment")
```

Comment:
The test seems to require the hostname found by the `Include` directive with inline comments, but the test doesn't seem to require that the words in the comment are excluded.

Reply:
the original fix is to discard inline comments after an `Include` directive

if the included file would contain `Host with_comment # to be ignored` then this patch would also require to update the `Host` directive parsing sed expression (which I already tried) only to find out it breaks https://github.com/scop/bash-completion/blob/main/test/fixtures/_known_hosts/config#L9 `Host = hus%%eth0 !negated  #not-a-comment` as until now fixtures considered inline comments as actual content to be parsed

This is precisely why I only patched the `Include` parsing expression and mentioned this future `Host` issue, because tests would fail one after the other

Reply:
I think the fixture for `Host` should also be updated.

----

By the way, I started this "conversation" (in the GitHub term) to discuss what should be tested. You've modified the test, but it doesn't seem to test the expectation by the original fix for `Include`. Could you test that a word that matches a filename in the comment is excluded? You can add another file in the fixture and put a hostname that should not be collected there. Then, you can include the filename in the inline comment on a `Include` line.

Reply:
I've again checked this conversation. In https://github.com/scop/bash-completion/pull/1199#discussion_r1625448295, I requested to add a regression test for the main purpose of this PR; i.e., we should test that "inside_comment" of `Include with_comment # inside_comment` is excluded from include targets.

However, the reply https://github.com/scop/bash-completion/pull/1199#discussion_r1626233739 talked about something completely different; it discussed a similar fix for Host to exclude the inline comments. To this, I replied (in the former half of https://github.com/scop/bash-completion/pull/1199#discussion_r1626965206) that `Host` should also be updated similarly, but I now think it doesn't need to be done within this PR.

In the latter half of https://github.com/scop/bash-completion/pull/1199#discussion_r1626965206, I pointed out that the test is not updated to include the regression test, but there is no further action from the original author of the PR.

Reply:
7aa420a2871cd17436fefd9fa8b468c7235bee47 for a regression test.